### PR TITLE
Include Ansible Kubernetes roles and playbooks to setup CentOS based K8s cluster.

### LINKED ISSUE #1135: Update the Ansible Kubernetes roles to include steps for CentOS
## Is this a BUG REPORT or FEATURE REQUEST?
FEATURE REQUEST

**What happened**:

Currently, the ansible K8s roles perform the Kubernetes setup on Ubuntu. These will need to support other operating systems like CentOS, CoreOS etc..,

A good reference to create multi-platform ansible roles is here : https://github.com/geerlingguy/ansible-role-apache/blob/master/tasks/main.yml

This issue will track the effort/changes needed for CentOS

### PR DESCRIPTION
**Pull request by dargasudarshan**
**What this PR does / why we need it**:
This PR will include the ansible roles and playbook to install required packages and setup CentOS based multinode  kubernetes cluster.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1135

**Special notes for your reviewer**:
Added a readme file K8s-CentOS.md with all the changes made for future reference.
@ksatchit Can you please review this PR?

### CODE REVIEW

**Comment by dargasudarshan** [comment]
TASK [k8s-hosts-centos : Setup k8s Hosts] ******************************************************************************************************************************
task path: /root/openebs/e2e/ansible/roles/k8s-hosts-centos/tasks/main.yml:127
NOTIFIED HANDLER Reload systemd
NOTIFIED HANDLER Restart kubelet
changed: [kubeminion01] => {"changed": true, "rc": 0, "stderr": "Shared connection to 172.28.128.32 closed.\r\n", "stdout": "WARN: Unknown option (ignored): --clusterip=10.96.0.1\r\nUpdating the host files...\r\nDisable swap\r\nSetting up the Minion using IPAddress: 172.28.128.32\r\nSetting up the Minion using Token: ff5a5b.ec54f28513d88a96\r\nSetting up the Minion using Discovery Token SHA: a8700478796e4fe87d44ce427cbf82784feed5978971d29660788a0b29ad96bd\r\n[kubeadm] WARNING: kubeadm is in beta, please do not use it for production clusters.\r\n[preflight] Running pre-flight checks\r\n[discovery] Trying to connect to API Server \"172.28.128.31:6443\"\r\n[discovery] Created cluster-info discovery client, requesting info from \"https://172.28.128.31:6443\"\r\n[discovery] Requesting info from \"https://172.28.128.31:6443\" again to validate TLS against the pinned public key\r\n[discovery] Cluster info signature and contents are valid and TLS certificate validates against pinned roots, will use API Server \"172.28.128.31:6443\"\r\n[discovery] Successfully established connection with API Server \"172.28.128.31:6443\"\r\n[bootstrap] Detected server version: v1.7.8\r\n[bootstrap] The server supports the Certificates API (certificates.k8s.io/v1beta1)\r\n\r\nNode join complete:\r\n* Certificate signing request sent to master and response\r\n  received.\r\n* Kubelet informed of new secure connection details.\r\n\r\nRun 'kubectl get nodes' on the master to see this machine join.\r\n", "stdout_lines": ["WARN: Unknown option (ignored): --clusterip=10.96.0.1", "Updating the host files...", "Disable swap", "Setting up the Minion using IPAddress: 172.28.128.32", "Setting up the Minion using Token: ff5a5b.ec54f28513d88a96", "Setting up the Minion using Discovery Token SHA: a8700478796e4fe87d44ce427cbf82784feed5978971d29660788a0b29ad96bd", "[kubeadm] WARNING: kubeadm is in beta, please do not use it for production clusters.", "[preflight] Running pre-flight checks", "[discovery] Trying to connect to API Server \"172.28.128.31:6443\"", "[discovery] Created cluster-info discovery client, requesting info from \"https://172.28.128.31:6443\"", "[discovery] Requesting info from \"https://172.28.128.31:6443\" again to validate TLS against the pinned public key", "[discovery] Cluster info signature and contents are valid and TLS certificate validates against pinned roots, will use API Server \"172.28.128.31:6443\"", "[discovery] Successfully established connection with API Server \"172.28.128.31:6443\"", "[bootstrap] Detected server version: v1.7.8", "[bootstrap] The server supports the Certificates API (certificates.k8s.io/v1beta1)", "", "Node join complete:", "* Certificate signing request sent to master and response", "  received.", "* Kubelet informed of new secure connection details.", "", "Run 'kubectl get nodes' on the master to see this machine join."]}
NOTIFIED HANDLER Reload systemd
NOTIFIED HANDLER Restart kubelet
changed: [kubeminion02] => {"changed": true, "rc": 0, "stderr": "Shared connection to 172.28.128.33 closed.\r\n", "stdout": "WARN: Unknown option (ignored): --clusterip=10.96.0.1\r\nUpdating the host files...\r\nDisable swap\r\nSetting up the Minion using IPAddress: 172.28.128.33\r\nSetting up the Minion using Token: ff5a5b.ec54f28513d88a96\r\nSetting up the Minion using Discovery Token SHA: a8700478796e4fe87d44ce427cbf82784feed5978971d29660788a0b29ad96bd\r\n[kubeadm] WARNING: kubeadm is in beta, please do not use it for production clusters.\r\n[preflight] Running pre-flight checks\r\n[discovery] Trying to connect to API Server \"172.28.128.31:6443\"\r\n[discovery] Created cluster-info discovery client, requesting info from \"https://172.28.128.31:6443\"\r\n[discovery] Requesting info from \"https://172.28.128.31:6443\" again to validate TLS against the pinned public key\r\n[discovery] Cluster info signature and contents are valid and TLS certificate validates against pinned roots, will use API Server \"172.28.128.31:6443\"\r\n[discovery] Successfully established connection with API Server \"172.28.128.31:6443\"\r\n[bootstrap] Detected server version: v1.7.8\r\n[bootstrap] The server supports the Certificates API (certificates.k8s.io/v1beta1)\r\n\r\nNode join complete:\r\n* Certificate signing request sent to master and response\r\n  received.\r\n* Kubelet informed of new secure connection details.\r\n\r\nRun 'kubectl get nodes' on the master to see this machine join.\r\n", "stdout_lines": ["WARN: Unknown option (ignored): --clusterip=10.96.0.1", "Updating the host files...", "Disable swap", "Setting up the Minion using IPAddress: 172.28.128.33", "Setting up the Minion using Token: ff5a5b.ec54f28513d88a96", "Setting up the Minion using Discovery Token SHA: a8700478796e4fe87d44ce427cbf82784feed5978971d29660788a0b29ad96bd", "[kubeadm] WARNING: kubeadm is in beta, please do not use it for production clusters.", "[preflight] Running pre-flight checks", "[discovery] Trying to connect to API Server \"172.28.128.31:6443\"", "[discovery] Created cluster-info discovery client, requesting info from \"https://172.28.128.31:6443\"", "[discovery] Requesting info from \"https://172.28.128.31:6443\" again to validate TLS against the pinned public key", "[discovery] Cluster info signature and contents are valid and TLS certificate validates against pinned roots, will use API Server \"172.28.128.31:6443\"", "[discovery] Successfully established connection with API Server \"172.28.128.31:6443\"", "[bootstrap] Detected server version: v1.7.8", "[bootstrap] The server supports the Certificates API (certificates.k8s.io/v1beta1)", "", "Node join complete:", "* Certificate signing request sent to master and response", "  received.", "* Kubelet informed of new secure connection details.", "", "Run 'kubectl get nodes' on the master to see this machine join."]}

RUNNING HANDLER [k8s-hosts : Restart kubelet] **************************************************************************************************************************
skipping: [kubeminion01] => {"changed": false, "skip_reason": "Conditional result was False"}
skipping: [kubeminion02] => {"changed": false, "skip_reason": "Conditional result was False"}
META: ran handlers
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************
kubeminion01               : ok=31   changed=16   unreachable=0    failed=0
kubeminion02               : ok=31   changed=16   unreachable=0    failed=0

File: e2e/ansible/roles/kubernetes-centos/tasks/1 [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
```
Comment (yudaykiran) [inline e2e/ansible/roles/kubernetes-centos/tasks/1]: What is this file? Is it a tmp file or main.yml got renamed?

File: e2e/ansible/roles/kubernetes-centos/1 [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (yudaykiran) [inline e2e/ansible/roles/kubernetes-centos/1]: What is this file? Is it a tmp file or main.yml got renamed?

File: e2e/ansible/roles/kubernetes-centos/tasks/1 (hunk 1 above)
Comment (dargasudarshan) [reply e2e/ansible/roles/kubernetes-centos/tasks/1]: Seems like a junk file. Removed it.

File: e2e/ansible/roles/k8s-hosts-centos/files/configure_k8s_host.sh [hunk 3]
```diff
@@ -0,0 +1,206 @@
+#!/bin/bash
```
Comment (yudaykiran) [inline e2e/ansible/roles/k8s-hosts-centos/files/configure_k8s_host.sh]: File not required.

File: e2e/ansible/roles/k8s-master-centos/files/configure_k8s_cni.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
```
Comment (yudaykiran) [inline e2e/ansible/roles/k8s-master-centos/files/configure_k8s_cni.sh]: File not required.

File: e2e/ansible/roles/k8s-master-centos/files/configure_k8s_master.sh [hunk 5]
```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
```
Comment (yudaykiran) [inline e2e/ansible/roles/k8s-master-centos/files/configure_k8s_master.sh]: File not required.

File: e2e/ansible/roles/k8s-master-centos/files/weave-daemonset-k8s-1.6.yaml [hunk 6]
```diff
@@ -0,0 +1,182 @@
+apiVersion: v1
```
Comment (yudaykiran) [inline e2e/ansible/roles/k8s-master-centos/files/weave-daemonset-k8s-1.6.yaml]: File not required.

**Comment by dargasudarshan** [comment]
@yudaykiran Removed the unwanted files.

**Pull request merged**